client/src/components/TrailShowPage.js

```diff
@@ -23,11 +29,42 @@ const TrailShowPage = (props) => {
 
   useEffect(() => {
     getTrail()
-  }, [])
+  }, [newReview])
+
+  const postReview = async (newReviewData) => {
+    const trailId = props.match.params.id
+    const reviewDataTrailId = { ...newReviewData, trailId: trailId }
+
+    try {
+      const response = await fetch("/api/v1/reviews", {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json",
+        }),
+        body: JSON.stringify(reviewDataTrailId),
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json()
+          const newErrors = translateServerErrors(body.errors)
+          return setErrors(newErrors)
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`
+          const error = new Error(errorMessage)
+          throw error
+        }
+      } else {
+        const body = await response.json()
+        setNewReview(body)
```

Comment:
This is creating an additional piece of state and an additional re-rendering of this component. Instead, we already have state for `trail.reviews`, so when we receive this new review back from the router, we want to update the `trail` state for reviews
```
setTrail({
  ...trail,
  reviews: [...trail.reviews, body]
})
```
Update the `trail` state, keep the existing trial object, and for the `reviews` property we want to keep the existing trail.reviews and concat on what is in `body`

Reply:
Done